File: packages/plugin-sdk-python/aisec_plugin/attack.py
"""Attack plugin base."""

from __future__ import annotations

from typing import Any

from aisec_plugin.base import Plugin, PluginContext


class AttackPlugin(Plugin):
    @classmethod
    def execute_attack(cls, ctx: PluginContext) -> dict[str, Any]:
        raise NotImplementedError

    @classmethod
    def setup_handlers(cls) -> None:
        cls.register("execute_attack", cls.execute_attack)

File: packages/plugin-sdk-js/src/attack.js
import { Plugin } from './base.js';

export class AttackPlugin extends Plugin {
  static setupHandlers() {
    this.register('execute_attack', (ctx) => this.executeAttack(ctx));
  }

  static executeAttack(_ctx) {
    throw new Error('executeAttack() not implemented');
  }
}
